x-pack/plugins/infra/public/pages/metrics/settings/source_configuration_settings.tsx

```diff
@@ -92,7 +88,12 @@ export const SourceConfigurationSettings = ({
     formState,
   ]);
 
-  const hasUnsavedChanges = isFormDirty || Object.keys(infraUiSettings.unsavedChanges).length > 0;
+  const unsavedChangesCount = Object.keys(getUnsavedChanges()).length;
+  const infraUiSettingsUnsavedChangesCount = Object.keys(infraUiSettings.unsavedChanges).length;
+  // Count changes from the feature section settings and general infra settings
+  const unsavedFormChangesCount = infraUiSettingsUnsavedChangesCount + unsavedChangesCount;
+
+  const isFormDirty = infraUiSettingsUnsavedChangesCount > 0 || unsavedChangesCount > 0;
```

Comment:
I needed to refactor this: So we need to get the changes from the settings form and the features section together to check if there are changes and to get the total changes count